tools/tiller-hack/tiller-hack.go

```diff
@@ -0,0 +1,232 @@
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	// "gopkg.in/yaml.v2"
+	// "github.com/ghodss/yaml"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
+
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/golang/protobuf/proto"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+	rspb "k8s.io/helm/pkg/proto/hapi/release"
+)
+
+var (
+	kubeconfig         = flag.String("kubeconfig", "", "Absolute path to the kubeconfig file.")
+	configMapName      = flag.String("configmap", "", "Name of the configmap to patch.")
+	configMapNamespace = flag.String("namespace", "kube-system", "Namespace of the configmap, defaults to \"kube-system\"")
+	restartTillerFlag  = flag.Bool("restart-tiller", false, "Restart tiller after patching the configmap")
+)
+
+var b64 = base64.StdEncoding
+
+// In older versions of helm, the compression of release information has not been used.
+// This header identify whether the following info is compressed or not.
+var magicGzip = []byte{0x1f, 0x8b, 0x08}
+
+// BuildClient returns kubernetes clientset
+func BuildClient(kubeconfig string) (*kubernetes.Clientset, error) {
+	var config *rest.Config
+	var err error
+
+	kubeconfigEnv := os.Getenv("KUBECONFIG")
+
+	if kubeconfigEnv != "" {
+		kubeconfig = kubeconfigEnv
+	}
+
+	if kubeconfig != "" {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		config, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		return nil, err
+	}
+	k8s, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return k8s, nil
+}
+
+func getConfigmapData(k8sClient *kubernetes.Clientset, configMapName, configMapNamespace string) ([]byte, error) {
+	cfgMap, err := k8sClient.Core().ConfigMaps(configMapNamespace).Get(configMapName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("fail to get configmap %s/%s with error: %+v", configMapNamespace, configMapName, err)
+	}
+	data, ok := cfgMap.Data["release"]
+	if !ok {
+		return nil, fmt.Errorf("fail to locate \"release\" key in configmap %s/%s", configMapNamespace, configMapName)
+	}
+	return []byte(data), nil
+}
+
+func makeConfigMapBackup(k8sClient *kubernetes.Clientset, configMapName, configMapNamespace string) error {
+	var err error
```

Comment:
no need to declare this explicitly as it's created below.

Reply:
done